src/components/singleValueAutocompleteInput.js

```diff
@@ -0,0 +1,468 @@
+    const [inputValue, setInputValue] = useState('');
+    const [interactionFilter, setInteractionFilter] = useState({});
+    const isNumberType = type === 'number';
+
+    const validPattern = pattern || null;
+    const validMinlength = minlength || null;
+    const validMaxlength = maxlength || null;
+    const validMinvalue = minvalue || null;
+
+    const patternMismatchMessage = useText(validationPatternMismatch || []);
+    const typeMismatchMessage = useText(validationTypeMismatch || []);
+    const valueMissingMessage = useText(validationValueMissing || []);
+    const tooLongMessage = useText(validationTooLong || []);
+    const tooShortMessage = useText(validationTooShort || []);
+    const belowMinimumMessage = useText(validationBelowMinimum || []);
+    const helperTextResolved = useText(helperTextRaw || []);
+
+    const validationMessage = (validityObject) => {
+      if (!validityObject) {
+        return '';
+      }
+      if (validityObject.customError && patternMismatchMessage) {
+        return patternMismatchMessage;
+      }
+      if (validityObject.valid) {
+        return '';
+      }
+      if (validityObject.typeMismatch && typeMismatchMessage) {
+        return typeMismatchMessage;
+      }
+      if (validityObject.patternMismatch && patternMismatchMessage) {
+        return patternMismatchMessage;
+      }
+      if (validityObject.valueMissing && valueMissingMessage) {
+        return valueMissingMessage;
+      }
+      if (validityObject.tooLong && tooLongMessage) {
+        return tooLongMessage;
+      }
+      if (validityObject.tooShort && tooShortMessage) {
+        return tooShortMessage;
+      }
+      if (validityObject.rangeUnderflow && belowMinimumMessage) {
+        return belowMinimumMessage;
+      }
+      return '';
+    };
+
+    const handleValidation = (validation) => {
+      if (validation) {
+        setErrorState(!validation.valid);
+      }
+      const message = validationMessage(validation) || helperTextResolved;
+      setHelper(message);
+    };
+
+    const customPatternValidation = (target) => {
+      const { value: eventValue, validity } = target;
+      if (!pattern) {
+        return validity;
+      }
+      const patternRegex = RegExp(`^${pattern}$`);
+      const isValid = patternRegex.test(eventValue);
+      target.setCustomValidity(isValid ? '' : 'Invalid field.');
+      return {
+        ...validity,
+        valid: isValid,
+        patternMismatch: !isValid,
+      };
+    };
+
+    B.defineFunction('Clear', () => {
+      setValue('');
+      setInputValue('');
+    });
+
+    /**
+     * @name Filter
+     * @param {Property} property
+     * @returns {Void}
+     */
+    B.defineFunction(
+      'Filter',
+      ({ event, property: propertyArg, interactionId }) => {
+        let eventValue;
+
+        if (event.target) {
+          eventValue = event.target.value;
+        } else if (event instanceof Date) {
+          eventValue = value.toISOString();
```

Comment:
This should be tested. I belive the backend will only accept mysql datetime formats.


```ts
const { DateFnsUtils } = window.MaterialUI;

const DateFns = new DateFnsUtils();

if (selectedDate) return DateFns.format(selectedDate, 'yyyy-MM-dd HH:mm:ss'):

return null

```

Reply:
Decided to remove the date related stuff because currently this autocomplete only supports list properties.